aptos-move/framework/aptos-framework/sources/ResourceAccount.move

```diff
@@ -73,6 +73,7 @@ module AptosFramework::ResourceAccount {
             optional_auth_key
         };
         Account::rotate_authentication_key_internal(&resource, auth_key);
+        resource
```

Comment:
iirc script functions can't return values.  If we want a return value, we make an inner function, and then we can call it here and elsewhere.

The concept behind them not returning values is that if you were to submit a transaction with just it, where does the output go?

Reply:
ooh I see, that's a good point!